source/client/RunManagement.py

```diff
@@ -0,0 +1,250 @@
+    groups_wilds = []
+    temp_run_group = []
+    aces_list =[]
+    gap_flag = False
+    for card in card_group:                    # separate unassigned wilds and Aces from rest of run.
+        # pull out wilds that have not been assigned (do this before Aces, in case rules changed so that Aces are wild).
+        if card.tempnumber in wild_numbers:
+            groups_wilds.append(card)
+            # pull out Aces that have not been assigned hi or low, but don't pull out Wilds assigned to low Ace slot.
+        elif card.tempnumber == 1 and card.number == 1:
+            aces_list.append(card)
+        else:
+            temp_run_group.append(card)
+    card_group  = []                         # rebuild card_group below
+    for card in temp_run_group:
+        if first_card:
+            first_card = False
+            card_group.append(card)
+        else:
+            abs_temp_number = abs(card_group[-1].tempnumber)  # Aces designated to be low have tempnumber = -1.
+            if card.tempnumber == (abs_temp_number + 1):
+                card_group.append(card)
+            elif card.tempnumber == (abs_temp_number + 2) and len(groups_wilds) > 0:
+                this_wild = groups_wilds.pop(0)
+                this_wild.tempnumber = abs_temp_number + 1
+                card_group.append(this_wild)
+                card_group.append(card)
+            elif card.tempnumber == abs_temp_number:
+                if isWild(card, wild_numbers):
+                    card.tempnumber = card.number  # reset Wild card back to original value (0 for Jokers)
+                    groups_wilds.append(card)
+                elif isWild(card_group[-1], wild_numbers):
+                    this_wild = card_group.pop(-1)
+                    this_wild.tempnumber = this_wild.number
+                    groups_wilds.append(this_wild)
+                    card_group.append(card)
+                else:
+                    raise Exception('Card value already in the run.')
+            elif card.tempnumber > (abs_temp_number + 2):
+                raise Exception('too big a gap between numbers')
+            else:
+                # gap must be 1 card. It's possible to free up a wild later in run then gap of 1 can be closed.
+                gap_flag =True
+    if gap_flag and len(groups_wilds) > 0:
+        temp_run_group = card_group
+        card_group = []
+        card_group.append(temp_run_group[0])
+        for indx in range(len(temp_run_group)-1):
+            card = temp_run_group[indx+1]
+            if temp_run_group[indx].tempnumber == card.tempnumber + 1:
+                card_group.append(card)
+            elif temp_run_group[indx].tempnumber == card.tempnumber + 2:
+                if len(groups_wilds) > 0:
+                    this_wild.pop(groups_wilds[0])
+                    card_group.append(this_wild)
+                    card_group.append(card)
+                else:
+                    raise Exception('too big a gap between numbers')
```

Comment:
This exception seems to actually be 'not enough wild cards' and not 'too big a gap between numbers' - I think you copy pasted

Reply:
Rephrased exception.